oauth2_http/java/com/google/auth/oauth2/IdentityPoolCredentialSource.java

```diff
@@ -80,9 +240,12 @@ public IdentityPoolCredentialSource(Map<String, Object> credentialSourceMap) {
     } else if (credentialSourceMap.containsKey("url")) {
       credentialLocation = (String) credentialSourceMap.get("url");
       credentialSourceType = IdentityPoolCredentialSourceType.URL;
+    } else if (credentialSourceMap.containsKey("certificate")) {
```

Comment:
nit: Can the if/else blocks use the boolean variables above? `filePresent` and the others?